group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -501,13 +521,11 @@ public void testOffsetDeleteWith(
 
             final OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection expectedResponsePartitionCollection =
                 new OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection();
-            if (hasOffset(groupId, topic, partition)) {
```

Comment:
why did we remove this check?

Reply:
This was actually wrong. The response should contain an entry for a partition even if the partition had no offset. See https://github.com/apache/kafka/pull/15221/files#diff-a4ad0e0a77c78dde2a841d055dd64e15f7da00b8a2a9e3279d74b718d5c612bbR607.

Reply:
Which one is the test where we verified we return a response even if the partition had no offset?

Reply:
Basically, all the tests using `testOffsetDeleteWith` (this method) verify it. I only had to remove the condition here in order to always expect the response.